src/d3/xy-chart.d3.ts

```diff
@@ -2,15 +2,14 @@ import * as d3 from 'd3';
 import { BaseD3, RenderOptions as BaseRenderOptions } from './base.d3';
 import { Observable } from 'rxjs';
 import { formatX } from '../utils/formatters';
+import { XYChartData, Datum } from '../datasets/types';
 
-export interface Datum {
-  date: Date;
-  value: number;
-}
+// TODO: Generic Datum type
+export type DatumDN = Datum<Date, number>;
```

Comment:
nit: probably not use abbreviation unless very common?

I can tell DN stands for Date and Number from its definition, but it would be hard to tell if I didn't know the definition. Also, because DatumDN inherits Datum, it makes more sense to keep `Datum` as suffix than prefix.

Reply:
I would like to let `XYChartD3` and `RenderOptions` take type variables for x and y, since it will later be extended by other charts like `BarChart`. However, we may also need to take different format functions (value to string) for different types. Do you have any suggestions for it?

Reply:
Hmm good point, should we have them as properties of `XYChartMeta`?

```ts
interface XYChartMeta<X, Y> extends ChartMeta<...> {
  formatX(value: X): string;
  formatY(value: Y): string;
}
```

Reply:
I think y will always be numeric while x should be generic.

Reply:
I think letting `XYChartMeta`  bring the formatters makes sense.

Reply:
This feature will be added in a separate PR.